File: src/Client/src/models/FoodItem.ts
import {NutritionalContent} from "./NutritionalContent.ts";
import {FoodItemForm} from "./FoodItemForm.ts";
import {v4 as UuidV4} from "uuid";

export class FoodItem {
    id: string;
    brand: string;
    productName: string;
    nutritionalContent: NutritionalContent;
    ownerId: string;

    constructor(id: string, brand: string, productName: string, nutritionalContent: NutritionalContent, ownerId: string) {
        this.id = id;
        this.brand = brand;
        this.productName = productName;
        this.nutritionalContent = nutritionalContent;
        this.ownerId = ownerId;
    }
}

export function postFoodItem(foodForm: FoodItemForm): FoodItem {
    // post stuff goes here
    const fid = UuidV4();
    const uid = UuidV4();
    return new FoodItem(
        fid,
        foodForm.brand,
        foodForm.productName,
        new NutritionalContent(
            foodForm.protein,
            foodForm.carbohydrate,
            foodForm.fat,
            foodForm.kCal,
            foodForm.unit),
        uid);
}

File: src/Domain/FoodItem.cs
using System;
using Domain.Common;

namespace Domain;

public class FoodItem : BaseEntity
{
    public string Brand { get; set; }
    public string ProductName { get; set; }
    public NutritionalContent NutritionalContent { get; set; }
    public Guid OwnerId { get; set; }


    // EF Core
    private FoodItem()
    {
    }

    public FoodItem(string brand, string productName, NutritionalContent nutritionalContent, Guid ownerId)
    {
        Id = Guid.NewGuid();
        Brand = brand;
        ProductName = productName;
        NutritionalContent = nutritionalContent;
        OwnerId = ownerId;
    }
}